### PR TITLE
feat: loop over JSON

### PR DESCRIPTION
**Pull request by mgeerse**
Loop over array items in a `JSON` (or a JSON like `Text` variable) object within the new action IDE.

Only thing missing from this pull request is a functioning test script. I can't figure out how to make testing work with the `steps()` function.

### CODE REVIEW

File: functions/json-loop/1.0/index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const jsonLoop = async ({ input, path }, steps) => {
+  try {
+    input = JSON.parse(input);
+  } finally {
+    if (path && path.includes('.')) {
+      path.split('.').forEach((key) => {
+        input = input[key];
+      });
+    } else if (typeof path !== 'undefined' && path !== '' && path !== null) {
```
Comment (justin-betty) [inline functions/json-loop/1.0/index.js:9]: could you not just check if path is there `typeof path !== 'undefined' && path`

**Comment by justin-betty** [comment]
Have a look in the native loop function test to see how to test if a child step is called

File: functions/json-loop/1.0/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const jsonLoop = async ({ input, path }, steps) => {
+  try {
+    input = JSON.parse(input);
+  } finally {
+    if (path && path.includes('.')) {
+      path.split('.').forEach((key) => {
+        input = input[key];
+      });
+    } else if (typeof path !== 'undefined' && path) {
```
Comment (justin-betty) [inline functions/json-loop/1.0/index.js:9]: After checking again `typeof path !== 'undefined'` is not needed just check if path is there:

`switch (path) {
  case path.includes('.'): {
    path.split('.').forEach((key) => {
      input = input[key];
    });
    break;
  }

  default:
    input = input[path];
}
`

File: functions/json-loop/1.0/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const jsonLoop = async ({ input, path }, steps) => {
+  try {
+    input = JSON.parse(input);
+  } finally {
+    if (path && path.includes('.')) {
```
Comment (justin-betty) [inline functions/json-loop/1.0/index.js:5]: you don't need the `else if`.

Just do 

```
if (path) {
  switch (true) {
    case path.includes('.'): {
      path.split('.').forEach((key) => {
        input = input[key];
      });
      break;
    }

    default:
      input = input[path];
  }
}

Comment (mgeerse) [reply functions/json-loop/1.0/index.js]: Nice remark! Thank you. Updating as we speak.

**Review by justin-betty — APPROVED** [review]

**Pull request merged**